opensrp-giz-malawi/src/main/java/org/smartregister/giz/dao/ReportDao.java

```diff
@@ -0,0 +1,419 @@
+            String vaccineCategory,
+            List<Vaccine> issuedVaccines
+    ) {
+        List<Alert> generatedAlerts = new ArrayList<>();
+        try {
+            if (vaccineSchedules != null && vaccineSchedules.containsKey(vaccineCategory)) {
+                for (VaccineSchedule curSchedule : vaccineSchedules.get(vaccineCategory).values()) {
+                    if (curSchedule == null) continue;
+                    Alert curAlert = curSchedule.getOfflineAlert(baseEntityId, dob.toDate(), issuedVaccines);
+                    if (curAlert != null && curAlert.startDate() != null) {
+                        generatedAlerts.add(curAlert);
+                    }
+                }
+            }
+            return generatedAlerts;
+        } catch (Exception e) {
+            return new ArrayList<>();
+        }
+    }
+
+    private static List<Alert> computeChildAlerts(int age, DateTime anchorDate, String baseEntityId, @Nullable List<Vaccine> issuedVaccines) {
+        try {
+            String category = age < 5 ? "child" : "child_over_5";
+            HashMap<String, HashMap<String, VaccineSchedule>> vaccineSchedules = getVaccineSchedules(category);
+            return getInMemoryAlerts(vaccineSchedules, baseEntityId, anchorDate, category, issuedVaccines == null ? new ArrayList<>() : issuedVaccines);
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+        return new ArrayList<>();
+    }
+
+    public static List<EligibleChild> fetchLiveEligibleChildrenReport(@Nullable String communityId, @Nullable Date dueDate) {
+        // fetch all children in the village
+        String _communityId = "('" + StringUtils.join(communityId, "','") + "')";
+        int days = Days.daysBetween(new DateTime().toLocalDate(), new DateTime(dueDate).toLocalDate()).getDays();
+        String sql = "select cd.base_entity_id , c.first_name , c.last_name family_name, c.middle_name ," +
+                "c.dob , c.gender , l.location_id " +
+                "from ec_child_details cd " +
+                "inner join ec_client c on cd.base_entity_id = c.base_entity_id and c.is_closed = 0 COLLATE NOCASE " +
+                "inner join ec_family_member_location l on l.base_entity_id = cd.base_entity_id COLLATE NOCASE";
+
+        if (communityId != null && !communityId.isEmpty())
+            sql += " and ( l.location_id IN " + _communityId + " or '" + communityId + "' = '') ";
+        sql += "order by c.first_name , c.last_name , c.middle_name ";
+        Map<String, List<Vaccine>> allVaccines = fetchAllVaccines();
+        List<EligibleChild> eligibleChildren = new ArrayList<>();
+        DataMap<Void> dataMap = getData(eligibleChildren, days, allVaccines, dueDate);
+        readData(sql, dataMap);
+        return eligibleChildren;
+    }
+
+    private static DataMap<Void> getData(List<EligibleChild> eligibleChildren, int days, Map<String, List<Vaccine>> allVaccines, Date dueDate) {
+
+        AbstractDao.DataMap<Void> dataMap = c -> {
```

Comment:
Could you give a more descriptive variable than `c`?

Reply:
Done